src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnArithmetic.cs

```diff
@@ -188,14 +188,24 @@ public void Modulo(bool scalar, PrimitiveColumnContainer<bool> column)
         }
         public void And(PrimitiveColumnContainer<bool> left, PrimitiveColumnContainer<bool> right)
         {
+            long index = 0;
             for (int b = 0; b < left.Buffers.Count; b++)
             {
                 var mutableBuffer = left.Buffers.GetOrCreateMutable(b);
                 var span = mutableBuffer.Span;
                 var otherSpan = right.Buffers[b].ReadOnlySpan;
                 for (int i = 0; i < span.Length; i++)
                 {
-                    span[i] = (bool)(span[i] & otherSpan[i]);
+                    if (BitmapHelper.IsValid(right.NullBitMapBuffers[b].ReadOnlySpan, i))
```

Comment:
Where is the "left" getting checked?

Reply:
Sorry - my brain is mostly thinking in the "Generic Math" version of DataFrame. I can't remember where the left is checked.

Reply:
Hello Jake. Left side validity bitmap is cloned from the original left side. So for null values from the left side it is also null. Only right side nulls were ignored (and caused division by zero exception during / operations)